### PR TITLE
[Feature] font size switch buttons completed

### PR DESCRIPTION
**Pull request by hanyulo**
function of font size changing of article is totally completed.

### CODE REVIEW

File: src/components/FontChangeButton.js [hunk 1]
```diff
@@ -1,35 +1,85 @@
...
+      case 'small':
+        fontStatusSmall = styles['font-status-active']
+        break
+      case 'medium':
+        fontStatusMedium = styles['font-status-active']
+        break
+      case 'large':
+        fontStatusLarge = styles['font-status-active']
+        break
+    }
```
Comment (nickhsine) [inline src/components/FontChangeButton.js]: Lines (30 to 64) should be refactored.

File: src/components/article/Introduction.js [hunk 2]
```diff
@@ -8,10 +8,18 @@ import React, { Component } from 'react'
...
   let content
   let block = false
   let introArr = []
+  let theStyle
+  if(fontSize ==='large') {
+    theStyle = styles['font-enlarge']
+  }else if(fontSize ==='medium') {
+    theStyle = styles['font-default']
+  }else {
+    theStyle = styles['font-shrink']
```
Comment (nickhsine) [inline src/components/article/Introduction.js]: Read [classnames](https://github.com/JedWatson/classnames) doc, and rewrite these lines.

File: src/components/FontChangeButton.scss [hunk 3]
```diff
@@ -4,25 +4,176 @@
...
+
+@-webkit-keyframes wrapper-hide { 0% {width: rem(133px);} 100% {width: 0;} }
+@-moz-keyframes    wrapper-hide { 0% {width: rem(133px);} 100% {width: 0;} }
+@keyframes         wrapper-hide { 0% {width: rem(133px);} 100% {width: 0;} }
+
+.animaStyle-wrapper-hide{
+  display:inline-block !important;
+  animation-name: wrapper-hide;
+  @include animated(0.4s);
 }
```
Comment (nickhsine) [inline src/components/FontChangeButton.scss]: Could all these keyframes for animation be replaced by animation mixins we wrote in ````src/themes/animation.scss```?

**Comment by hanyulo** [comment]
I have modified the code based on your comment. Please checkout my latest modification.

File: src/components/FontChangeButton.js [hunk 4]
```diff
@@ -1,35 +1,60 @@
...
+
   render() {
-    let animaStyle = styles['animaStyle-hide']
-    if(this.state.active ===true) {
-      animaStyle = styles['animaStyle-hide']
-    }else {
-      animaStyle = styles['animaStyle-show']
+    // font animation
+    let animaWrapper, animaButton, controlButtonIcon=fontIconClose
+    if(this.state.active===false && this.state.initialize===false) {
```
Comment (nickhsine) [inline src/components/FontChangeButton.js]: ```!this.state.active && !this.state.initialize```

File: src/components/FontChangeButton.js [hunk 5]
```diff
@@ -1,35 +1,60 @@
...
+      controlButtonIcon = fontIconClose
+      animaWrapper =   animaWrapper = styles['animaStyle-wrapper-hide']
+      animaButton = styles['anima-hide']
+    }else if(this.state.active===true && this.state.initialize===false) {
```
Comment (nickhsine) [inline src/components/FontChangeButton.js]: ```this.state.active && !this.state.initialize```

**Comment by nickhsine** [comment]
Btw, please beautify the codes before commit

File: src/components/FontChangeButton.js [hunk 6]
```diff
@@ -1,35 +1,60 @@
...
+       </div>
+       <div className={classNames(animaWrapper,styles['wrapper'])}></div>
+       <div onClick={()=>this.changeFontSize('large')} className={classNames(styles['image-container'],animaButton,styles['image-container-large'])}>
+         <img className={classNames(styles['image-large'],fontSize === 'large' ? styles['font-status-active'] : styles['font-status-inactive'])} src={fontIconLarge}/>
+       </div>
+       <div onClick={()=>this.changeFontSize('medium')} className={classNames(styles['image-container'],animaButton,styles['image-container-medium'])}>
+         <img className={classNames(styles['image-medium'],fontSize === 'medium' ? styles['font-status-active'] : styles['font-status-inactive'])} src={fontIconMedium}/>
+       </div>
+       <div onClick={()=>this.changeFontSize('small')} className={classNames(styles['image-container'],animaButton,styles['image-container-small'])}>
+         <img  className={classNames(styles['image-small'],fontSize === 'small' ? styles['font-status-active'] : styles['font-status-inactive'])} src={fontIconSmall}/>
```
Comment (nickhsine) [inline src/components/FontChangeButton.js]: could you give these img tags for buttons the ```styles['font-status-inactive']``` style as a part of default styling?
So you will not need write ```styles['font-status-inactive']```, just need  ```styles['font-status-active']```

File: src/components/FontChangeButton.scss [hunk 7]
```diff
@@ -4,25 +4,112 @@
...
+  @include fade-in(0.7s);
+}
+
+.animaStyle-wrapper-show{
+  display:inline-block !important;
+  @include expand-right(0.4s,133px);
+}
+
+@-webkit-keyframes local-fade-out { 0% {opacity: 1;} 100% {opacity: 0; visibility:hidden;} }
+@-moz-keyframes    local-fade-out { 0% {opacity: 1;} 100% {opacity: 0; visibility:hidden;} }
```
Comment (nickhsine) [inline src/components/FontChangeButton.scss]: Have you try to remove these vendor-prefixed properties? I guess maybe webpack will generate  this vendor-prefixed properties.

File: src/components/article/Introduction.js [hunk 8]
```diff
@@ -1,18 +1,24 @@
 /*eslint no-unused-vars:0*/
 'use strict'
-import classNames from 'classnames'
+import classNames from 'classnames/bind'
```
Comment (nickhsine) [inline src/components/article/Introduction.js]: why ```classnames/bind``` ?

File: src/containers/Article.js [hunk 9]
```diff
@@ -146,15 +146,18 @@ class Article extends Component {
     // for requestAnimationFrame
     this._ticking = false
     this.state = {
-      fontSize:'medium'
+      fontSize:'medium',
+      fontSizeSetted:false
```
Comment (nickhsine) [inline src/containers/Article.js]: ```isFontSizeSet```

File: src/containers/Article.js [hunk 10]
```diff
@@ -163,6 +166,13 @@ class Article extends Component {
     window.addEventListener('resize', this._setArticleBounding)
     // detect sroll position
     window.addEventListener('scroll', this._onScroll)
+    let storedfontSize = localStorage.getItem('fontSize')
+    if(storedfontSize !== null && this.state.fontSizeSetted===false) {
```
Comment (nickhsine) [inline src/containers/Article.js]: ```!this.state.isFontSizeSet```

**Comment by hanyulo** [comment]
All done!   :)

File: src/components/FontChangeButton.js [hunk 11]
```diff
@@ -1,35 +1,61 @@
...
+    let fontSize = this.props.fontSize
+    let buttonsSet = [ { type: 'large', icon: fontIconLarge },
+    { type: 'medium', icon: fontIconMedium },
+    { type: 'small', icon: fontIconSmall } ].map(function (button,i) {
+      return (
+        <div key={i} onClick={()=>{this.changeFontSize(button.type)}} className={cx(styles['image-container'], animaButton, styles['image-container-' + button.type])}>
+          <img className={cx(styles['image-' + button.type], fontSize === button.type ? styles['font-status-active'] : null)} src={button.icon}/>
+        </div>
+      )
+    }.bind(this))
```
Comment (nickhsine) [inline src/components/FontChangeButton.js]: Nice!

**Comment by nickhsine** [comment]
👍 💯 :shipit:

**Pull request merged**